lab_2_retrieval_w_bm25/main.py

```diff
@@ -47,6 +71,15 @@ def build_vocabulary(documents: list[list[str]]) -> list[str] | None:
 
     In case of corrupt input arguments, None is returned.
     """
+    if not isinstance(documents,list) or not\
+            all(isinstance(document,list) for document in documents):
+        return None
+    vocabulary = []
+    for document in documents:
+        if not all(isinstance(token,str) for token in document):
+            return None
+        vocabulary += (word for word in document if word not in vocabulary)
```

Comment:
is it possible to do a list comprehension here?

Reply:
i think I did now?